### ISSUE TITLE
Using scipy's genetic algorithm for initial parameter estimation in gradient descent 

### ISSUE THREAD

**Issue by zunzun** [beaupletga/Linear-Regression-Optimizers#1]
In your Python code for gradient descent regression, you use random values as initial parameters for the gradient descent's starting point. This can be suboptimal, and in more complex equations often results in the algorithm finding a local minimum. For this reason, the authors of scipy added a genetic algorithm for initial parameter estimation. The module is named scipy.optimize.differential_evolution.

I have used scipy's Differential Evolution genetic algorithm to determine initial parameters for fitting a double Lorentzian peak equation to Raman spectroscopy data and found that the results were excellent. The GitHub project, with a test spectroscopy data file, is:

https://github.com/zunzun/RamanSpectroscopyFit

If you have any questions, please let me know. My background is in nuclear engineering and industrial radiation physics, and I love Python so I will be glad to help.

James Phillips

**Comment by beaupletga** [comment]
Hi !

You'r absolutely right. I wanted to pick randomly a and b just to see the behavior of each algorithm. Of course, the algorithm with the preprocessing might be better because the preprocess makes values in [0,1] (and a and b are randomly set between 0 and 1).

However, I've just used what you advised me and made a new function, and it's working pretty good, even with 10 iter.
Thank you very much ! I've learned something :+1:
